### PR TITLE
add(sanitization/actions): deluge-labeling (injection) and torrent filename (injection/save)

### PR DESCRIPTION
**Pull request by zakkarry**
This PR adds some string sanitization to both the injection and saving of torrents where the name is used. This applies to some trackers that don't use normal file/folder structure and naming for their .torrents and then causes cross-seed to ENOENT on injection or incorrectly save.

Also addresses injecting into Deluge when the source is a torrent already in a '.cross-seed' label, will now check for the '.cross-seed' tag and not duplicate it to '.cross-seed.cross-seed'

### CODE REVIEW

File: src/utils.ts
Comment (mmgoodnow) [inline src/utils.ts:120]: I don't think filenames can actually contain slashes - where are these torrent names coming from and how is it possible they have slashes?

https://stackoverflow.com/questions/9847288/is-it-possible-to-use-in-a-filename

**Review by mmgoodnow — COMMENTED** [review]
`.cross-seed` de-nesting stuff looks good, but I'm a little confused about the filenames

File: src/utils.ts
Comment (zakkarry) [reply src/utils.ts:120]: It's not the filename - it's the 'name' in the torrent file, which is what cross-seed is using to create the filename for the .torrent file.

I didn't think it was possible either, but I had the guy send me a screenshot of the torrent file in a torrent file editor, and it did have that in the name, was a season pack, and the files were shown in the root of the torrent, not in a folder.

When he loaded into his client, it went to the folder layout in qBittorrent though, I just applied same logic to Deluge as well.

Comment (mmgoodnow) [reply src/utils.ts:120]: could we put this as a method on Metafile? `getFileSystemSafeName()` or something like that?

**Review by mmgoodnow — APPROVED** [review]
LGTM pending the open thread. I can cut a release later tonight once you merge

File: src/utils.ts
Comment (zakkarry) [reply src/utils.ts]: Moved it to Metafile

File: src/parseTorrent.ts
Comment (mmgoodnow) [inline src/parseTorrent.ts:113]: I don't think this needs to be static, it can be an instance method—then we can return `this.name.replace(...)`

File: src/torrent.ts
Comment (mmgoodnow) [inline src/torrent.ts:114]: will this `stripExtension` bit still work?

Comment (zakkarry) [reply src/torrent.ts:114]: You mean if the name has a / in it?

I don't think it will strip it....considering it wont endWith just the extension. But considering that if it ends with a / it wont have an extension, I don't think we would technically really need to worry about that? It's not a single file torrent.

Thoughts?

File: src/parseTorrent.ts
Comment (zakkarry) [reply src/parseTorrent.ts]: Updated.

File: src/torrent.ts
Comment (mmgoodnow) [reply src/torrent.ts:114]: makes sense

File: src/torrent.ts
Comment (mmgoodnow) [inline src/torrent.ts:115]: looks like we lost the `stripExtension` bit:

```suggestion
	const filename = `[${tag}][${tracker}]${stripExtension(meta.getFileSystemSafeName())}.torrent`;
```

Comment (zakkarry) [reply src/torrent.ts:115]: Oops, good eyes

Comment (zakkarry) [reply src/torrent.ts:115]: Ah good catch, yea that would have been lost after implementing it into Metafile.

**Review by mmgoodnow — APPROVED** [review]

**Pull request merged**
